ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts

```diff
@@ -509,34 +510,33 @@ export class AddressEmbeddedComponent implements OnDestroy, OnInit {
             tap((errorMessage) => this.stateUpdates$.next({formErrorMessages: [errorMessage]}))
         );
 
-        const removeTempAddressOperator = () => (val$: Observable<any>) => val$.pipe(
-            withLatestFrom(this.state$),
-            filter(([_, state]) => !!state.activityInstanceGuid),
-            tap(() => busyCounter$.next(1)),
-            concatMap(([_, state]) => this.addressService.deleteTempAddress(state.activityInstanceGuid)),
-            catchError(() => {
-                this.logger.logDebug(this.LOG_SOURCE, 'Temp delete failed. This is OK.');
-                return of(null);
-            }),
-            tap(() => busyCounter$.next(-1))
-        );
-
         const canSaveRealAddress = (address: Address | null) =>
             this.enoughDataToSave(address) && this.meetsActivityRequirements(address);
 
-        // "Real" as opposed to "Temp"
+        // "Real" as opposed to "Temp". Important we return the saved address as properties are added on server
         const saveRealAddressAction$ = this.saveTrigger$.pipe(
             withLatestFrom(this.formErrorMessages$, currentAddress$),
             filter(([_, formErrors, addressToSave]) => {
                 return !formErrors?.length && canSaveRealAddress(addressToSave);
             }),
             tap(() => busyCounter$.next(1)),
             concatMap(([_, formErrors, addressToSave]) => this.addressService.saveAddress(addressToSave, false)),
-            removeTempAddressOperator(),
             tap(() => busyCounter$.next(-1)),
             share()
         );
 
+        const deleteTempAddressAction$ = saveRealAddressAction$.pipe(
+            withLatestFrom(this.state$),
+            filter(([_, state]) => !!state.activityInstanceGuid),
+            tap(() => busyCounter$.next(1)),
+            concatMap(([_, state]) => this.addressService.deleteTempAddress(state.activityInstanceGuid)),
+            catchError(() => {
+                this.logger.logDebug(this.LOG_SOURCE, 'Temp delete failed. This is OK.');
+                return of(null);
+            }),
+            tap(() => busyCounter$.next(-1))
+        );
+
```

Comment:
We did want to delete the temp address if the real mailing address is saved (this is just for cleanup purpose), so deletion factored into its own action. Triggered by saveRealAddressAction$ observable emitting after completion of its action.